server/paymentProviders/stripe/virtual-cards.js

```diff
@@ -86,9 +86,21 @@ export const processAuthorization = async (stripeAuthorization, stripeEvent) =>
 
   await checkStripeEvent(host, stripeEvent);
 
-  // TODO : convert balance to the same currency as amount
-  const amount = convertToStripeAmount(host.currency, stripeAuthorization.amount);
-  const balance = await host.getBalanceWithBlockedFundsAmount();
+  const existingExpense = await models.Expense.findOne({
+    where: {
+      VirtualCardId: virtualCard.id,
+      data: { authorizationId: stripeAuthorization.id },
+    },
+  });
+
+  if (existingExpense) {
+    logger.warn(`Virtual Card authorization already reconciled, ignoring it: ${stripeAuthorization.id}`);
+    return;
+  }
+
+  const currency = stripeAuthorization.currency.toUpperCase();
+  const amount = convertToStripeAmount(currency, Math.abs(stripeAuthorization.amount));
+  const balance = await host.getBalanceWithBlockedFundsAmount({ currency });
```

Comment:
Ouch! I haven't spotted this before, but we don't care about the balance of the host, we care about the balance of the collective. This is really important.